rai-sdk/src/main/java/com/relationalai/DefaultPollInterval.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 RelationalAI, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"): you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.relationalai;
+
+import org.awaitility.core.DurationFactory;
+import org.awaitility.pollinterval.PollInterval;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+public class DefaultPollInterval implements PollInterval {
```

Comment:
I'd call it `TransactionExecutePollInterval` (or similar) since it makes sense only within the context of transaction executions.

Reply:
Actually this could be also used out of the transaction scope for example waiting for an engine to be provisioned. I'm calling the class `DefaultPollInterval` because this is the strategy we want to poll with across all sdks and custom poll interval implementation should be different from this one